jobs/bump-lockfile.Jenkinsfile

```
@@ -35,89 +35,110 @@ try { lock(resource: "bump-${params.STREAM}") { timeout(time: 120, unit: 'MINUTE
 
-    stage("Build Metal") {
-        shwrap("cosa buildextend-metal")
-        shwrap("cosa buildextend-metal4k")
-    }
+        stage("Build Metal") {
+            shwrap("cosa buildextend-metal")
+            shwrap("cosa buildextend-metal4k")
+        }
 
-    stage("Build Live") {
-        shwrap("cosa buildextend-live --fast")
-        // Test metal4k with an uncompressed image and metal with a
-        // compressed one
-        shwrap("cosa compress --artifact=metal")
-    }
+        stage("Build Live") {
+            shwrap("cosa buildextend-live --fast")
+            // Test metal4k with an uncompressed image and metal with a
+            // compressed one
+            shwrap("cosa compress --artifact=metal")
+        }
 
-    try {
-        parallel metal: {
-            shwrap("kola testiso -S --scenarios pxe-install,iso-install,iso-offline-install --output-dir tmp/kola-testiso-metal")
-        }, metal4k: {
-            shwrap("kola testiso -S --scenarios iso-install,iso-offline-install --qemu-native-4k --output-dir tmp/kola-testiso-metal4k")
+        try {
+            parallel metal: {
+                shwrap("kola testiso -S --scenarios pxe-install,iso-install,iso-offline-install --output-dir tmp/kola-testiso-metal")
+            }, metal4k: {
+                shwrap("kola testiso -S --scenarios iso-install,iso-offline-install --qemu-native-4k --output-dir tmp/kola-testiso-metal4k")
+            }
+        } finally {
+            shwrap("tar -cf - tmp/kola-testiso-metal/ | xz -c9 > ${env.WORKSPACE}/kola-testiso-metal.tar.xz")
+            shwrap("tar -cf - tmp/kola-testiso-metal4k/ | xz -c9 > ${env.WORKSPACE}/kola-testiso-metal4k.tar.xz")
+            archiveArtifacts allowEmptyArchive: true, artifacts: 'kola-testiso*.tar.xz'
         }
-    } finally {
-        shwrap("tar -cf - tmp/kola-testiso-metal/ | xz -c9 > ${env.WORKSPACE}/kola-testiso-metal.tar.xz")
-        shwrap("tar -cf - tmp/kola-testiso-metal4k/ | xz -c9 > ${env.WORKSPACE}/kola-testiso-metal4k.tar.xz")
-        archiveArtifacts allowEmptyArchive: true, artifacts: 'kola-testiso*.tar.xz'
     }
 
-    // OK, it passed kola: just push to the branch. In the future, we might be
+    // OK, we're ready to push: just push to the branch. In the future, we might be
     // fancier here; e.g. if tests fail, just open a PR, or if tests passed but a
     // package was added or removed.
     stage("Push") {
-        shwrap("git -C src/config commit -am 'lockfiles: bump to latest' -m 'Job URL: ${env.BUILD_URL}' -m 'Job definition: https://github.com/coreos/fedora-coreos-pipeline/blob/main/jobs/bump-lockfile.Jenkinsfile'")
+        def message="lockfiles: bump to latest"
+        if (timestampOnly) {
+            message="lockfiles: bump timestamp"
+        }
+        shwrap("git -C src/config commit -am '$message' -m 'Job URL: ${env.BUILD_URL}' -m 'Job definition: https://github.com/coreos/fedora-coreos-pipeline/blob/main/jobs/bump-lockfile.Jenkinsfile'")
```

I think this might work, but I'd go with

```suggestion
        shwrap("git -C src/config commit -am '${message}' -m 'Job URL: ${env.BUILD_URL}' -m 'Job definition: https://github.com/coreos/fedora-coreos-pipeline/blob/main/jobs/bump-lockfile.Jenkinsfile'")
```

to be safe.

http://groovy-lang.org/syntax.html#_string_interpolation says:

> The placeholder expressions are surrounded by ${}. The curly braces may be omitted for unambiguous dotted expressions, i.e. we can use just a $ prefix in those cases.

But here we're not using a dotted expression, so I'm not sure if it'll still interpolate or not.

---

fixed